File: src/main/java/org/visual/module/RootModule.java
package org.visual.module;

import com.google.inject.AbstractModule;
import lombok.extern.slf4j.Slf4j;
import org.visual.local.store.VisualLocalStoreModule;

@Slf4j
public class RootModule extends AbstractModule {
  @Override
  protected void configure() {
    install(new VisualLocalStoreModule());
  }
}

File: src/main/kotlin/org/visual/module/RootModule.kt
package org.visual.module

import com.google.common.util.concurrent.ThreadFactoryBuilder
import com.google.inject.AbstractModule
import com.google.inject.Provides
import org.visual.shared.OS
import java.util.concurrent.ArrayBlockingQueue
import java.util.concurrent.ThreadPoolExecutor
import java.util.concurrent.ThreadPoolExecutor.CallerRunsPolicy
import java.util.concurrent.TimeUnit

class RootModule:AbstractModule() {

  override fun configure() {

  }

  @Provides
  fun executor(): ThreadPoolExecutor {
    return ThreadPoolExecutor(
      OS.cpuCore,
      OS.cpuCore + 1,
      1,
      TimeUnit.MINUTES,
      ArrayBlockingQueue(200),
      ThreadFactoryBuilder().setDaemon(true).setNameFormat("visual-model-%s").build(),
      CallerRunsPolicy()
    )
  }
}